lib/elf/file.rb

```diff
@@ -374,12 +374,16 @@ def is_compatible(other)
       raise TypeError.new("wrong argument type #{other.class} (expected Elf::File)") unless
         other.is_a? Elf::File
 
+      compatible_abi = case
+        when [@abi, other.abi].include?(Elf::OsAbi::SysV); true
```

Comment:
Not all ABIs are compatible with SysV though, can you please make this an explicit SysV/Linux and SysV/FreeBSD check?